File: de.raphaelmuesseler.financer.client.javafx/src/main/java/de/raphaelmuesseler/financer/client/javafx/dialogs/FinancerConfirmDialog.java
package de.raphaelmuesseler.financer.client.javafx.dialogs;

import javafx.scene.control.Label;
import javafx.scene.layout.Region;

public class FinancerConfirmDialog extends FinancerDialog<Boolean> {
    private Label questionLabel;
    private String question;

    public FinancerConfirmDialog(String question) {
        super(false);
        this.question = question;

        this.prepareDialogContent();
    }

    @Override
    protected Region getDialogContent() {
        this.questionLabel = new Label();
        return this.questionLabel;
    }

    @Override
    protected void prepareDialogContent() {
        this.questionLabel.setText(this.question);
    }

    @Override
    protected boolean checkConsistency() {
        return true;
    }

    @Override
    protected void onConfirm() {
        this.setValue(true);
        super.onConfirm();
    }

    @Override
    protected void onCancel() {
        this.setValue(false);
        super.onCancel();
    }
}

File: java/org.financer.client.javafx/src/main/kotlin/org/financer/client/javafx/dialogs/FinancerConfirmDialog.kt
package org.financer.client.javafx.dialogs

import com.jfoenix.controls.JFXDialogLayout
import javafx.scene.control.Label
import javafx.scene.layout.Region

open class FinancerConfirmDialog(private val question: String) : FinancerDialog<Boolean?>(false) {
    private var questionLabel: Label? = null
    override val dialogContent: Region?
        get() {
            questionLabel = Label()
            return questionLabel
        }

    override fun prepareDialogContent() {
        questionLabel!!.text = question
        (content as JFXDialogLayout).actions[0].requestFocus()
    }

    override fun checkConsistency(): Boolean {
        return true
    }

    override fun onConfirm() {
        value = true
        super.onConfirm()
    }

    override fun onCancel() {
        value = false
        super.onCancel()
    }

    init {
        prepareDialogContent()
    }
}